sdk/storage/azure-storage-common/src/test/java/com/azure/storage/common/implementation/StorageCrc64CalculatorTests.java

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.common.implementation;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+import java.util.Random;
+import java.util.stream.IntStream;
+import java.math.BigInteger;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class StorageCrc64CalculatorTests {
+    private static final int KB = 1024;
+    private static final String HEX_DATA = "C8E11B40D793D1526018"; // '\\xC8\\xE1\\x1B\\x40\\xD7\\x93\\xD1\\x52\\x60\\x18'
+    private static final byte[] HEX_BYTES = hexStringToByteArray();
+
+    @ParameterizedTest
+    @CsvSource({
+        "'', 0, 0",
+        "'Hello World!', 0, 208604604655264165",
+        "'123456789!@#$%^&*()', 0, 2153758901452455624",
+        "'This is a test where the data is longer than 64 characters so that we can test that code path.', 0, 2736107658526394369", })
+    void testCompute(String data, long initial, long expected) {
+        byte[] bytes = data.getBytes();
+        long actual = StorageCrc64Calculator.compute(bytes, initial);
+        assertEquals(expected, actual);
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+        "0, 3386042136331673945",
+        "208604604655264165, 4570059697646401418",
+        "2153758901452455624, 13366433516720813220",
+        "12345, 5139183895903464380" })
+    void testComputeWithBinaryData(long initial, String expectedStr) {
+        //byte[] hexBytes = hexStringToByteArray(hexData);
+        long expected = unsignedLongFromString(expectedStr);
+        long actual = StorageCrc64Calculator.compute(HEX_BYTES, initial);
+        assertEquals(expected, actual);
+    }
+
+    private static byte[] hexStringToByteArray() {
+        //int length = StorageCrc64CalculatorTests.hexData.length();
+        int length = HEX_DATA.length();
+        byte[] data = new byte[length / 2];
+        for (int i = 0; i < length; i += 2) {
+            data[i / 2] = (byte) ((Character.digit(StorageCrc64CalculatorTests.HEX_DATA.charAt(i), 16) << 4)
+                + Character.digit(StorageCrc64CalculatorTests.HEX_DATA.charAt(i + 1), 16));
+        }
+        return data;
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = { 2, 3, 10 })
+    void testCompose(int numSegments) {
```

Comment:
I'm wondering if we can combine the `testCompose` and `testComposeDifferingBlockSizes` to a single test case where we parameterize based on numSegment,minBlockSize,MaxBlockSize? For these cases, we would just make the minBlockSize and maxBlockSize equal to `KB`. That should help cut down in the amount of code needed for these and make it easier to add future cases.